pkg/core/block/block.go

```diff
@@ -92,12 +96,18 @@ func NewBlockFromTrimmedBytes(b []byte) (*Block, error) {
 
 	block.Script.DecodeBinary(br)
 
-	lenTX := br.ReadVarUint()
-	block.Transactions = make([]*transaction.Transaction, lenTX)
-	for i := 0; i < int(lenTX); i++ {
-		var hash util.Uint256
-		hash.DecodeBinary(br)
-		block.Transactions[i] = transaction.NewTrimmedTX(hash)
+	lenHashes := br.ReadVarUint()
+	if lenHashes > 0 {
+		var consensusDataHash util.Uint256
+		consensusDataHash.DecodeBinary(br)
```

Comment:
In order to match C#, we have to store and serialize all hashes (ConsensusData and txes). They store these hashes together in `TrimmedBlock.Hashes` array, see https://github.com/neo-project/neo/blob/master/src/neo/Network/P2P/Payloads/Block.cs#L137

Reply:
Yeah, I've seen that, still can't figure out what's the point, really. Headers don't have any hashes, blocks have full consensus data. It can be seen as some kind of cache, but it's not used in any way, you read it and then just throw away. And this is DB layer (trimmed blocks are only stored there), we can deviate from C# here (we can't at formats used to transmit things over the wire).